core/src/main/java/org/elasticsearch/action/ingest/SimulatePipelineTransportAction.java

```diff
@@ -56,6 +57,10 @@ protected void doExecute(SimulatePipelineRequest request, ActionListener<Simulat
             } else {
                 simulateRequest = SimulatePipelineRequest.parse(source, request.isVerbose(), pipelineStore);
             }
+        } catch (PipelineFactoryError e) {
+            // TODO(talevy): wrap SimulatePipelineResponse & PipelineFactoryErrorActionResponse into ONE object so this would work
+            listener.onResponse(new PipelineFactoryErrorActionResponse(e));
```

Comment:
maybe `SimulatePipelineResponse` can just hold a `PipelineFactoryError` field? Then this would work and be simple?